File: rick-morty-app/src/app/interfaces/episode.ts
import { Character } from "./character";

export interface Episode {
  id:          number;
  name:        string;
  airDate:     string;
  episodeCode: string;
  characters:  Character[];
  url:         string;
  created:     Date;
}

export interface EpisodeResponse {
  results: Episode[];
  currentPage: number;
  next?: string;
  prev?: string;
}

File: BFFRickAndMorty/Models/Episode.cs
using System.Text.Json.Serialization;

namespace BFFRickAndMorty.Models
{
    public class Episode
    {
        [JsonPropertyName("id")]
        public long Id { get; set; }

        [JsonPropertyName("name")]
        public string Name { get; set; }

        [JsonPropertyName("air_date")]
        public string AirDate { get; set; }

        [JsonPropertyName("episode")]
        public string EpisodeCode { get; set; }

        [JsonPropertyName("characters")]
        public List<string> CharacterUrls { get; set; } = new();

        [JsonIgnore]
        public List<Character> Characters { get; set; } = new();

        [JsonPropertyName("url")]
        public string Url { get; set; }

        [JsonPropertyName("created")]
        public DateTimeOffset Created { get; set; }
    }
}
